### PR TITLE
Delete secrets when downloading

### PR DESCRIPTION
**Pull request by idrysdale**
I've decided to delete a secret on the server whether it successfully
decrypts or not. I'm not sure what good it'd be to keep it on the server if
it's failed to decrypt?

Also, if we fail to either decode the metadata (for the UUID), or to
actually delete the secret on the server I've opted to log this.

### CODE REVIEW

File: fluidkeys/secretreceive.go [hunk 1]
```diff
@@ -85,6 +88,16 @@ func downloadAndDecryptSecrets(key pgpkey.PgpKey) (decryptedSecrets []string, se
 		} else {
 			decryptedSecrets = append(decryptedSecrets, decryptedContent)
 		}
+		jsonMetadata, err := decrypt(encryptedSecret.EncryptedMetadata, privateKey)
```
Comment (paulfurley) [inline fluidkeys/secretreceive.go]: missing an `err` check here

Comment (idrysdale) [reply fluidkeys/secretreceive.go]: Good spot. Fixed!

**Comment by paulfurley** [comment]
I've just been playing with this and the fact that it doesn't prompt caught me out: my first key got secrets OK, but then it prompted for a password on the second one. While I was messing around looking for the password, I think it deleted the secrets for the first one, since they weren't there when I ran it again!

It's a little disorientating having it (silently) delete secrets while at the same time prompting for passworsd

**Comment by paulfurley** [comment]
I'm gonna merge to crack on, but add a card that we could do this with a prompt. Nice one! :)

**Pull request merged**